llama_index/indices/managed/google/generativeai/base.py

```diff
@@ -0,0 +1,240 @@
+        """Returns the corpus ID being used by this GoogleIndex."""
+        return self._store.corpus_id
+
+    def _insert(self, nodes: Sequence[BaseNode], **insert_kwargs: Any) -> None:
+        """Inserts a set of nodes."""
+        self._index.insert_nodes(nodes=nodes, **insert_kwargs)
+
+    def insert_documents(self, documents: Sequence[Document], **kwargs: Any) -> None:
+        """Inserts a set of documents."""
+        for document in documents:
+            self.insert(document=document, **kwargs)
+
+    def delete_ref_doc(
+        self, ref_doc_id: str, delete_from_docstore: bool = False, **delete_kwargs: Any
+    ) -> None:
+        """Deletes a document and its nodes by using ref_doc_id."""
+        self._index.delete_ref_doc(ref_doc_id=ref_doc_id, **delete_kwargs)
+
+    def update_ref_doc(self, document: Document, **update_kwargs: Any) -> None:
+        """Updates a document and its corresponding nodes."""
+        self._index.update(document=document, **update_kwargs)
+
+    def as_retriever(self, **kwargs: Any) -> BaseRetriever:
+        """Returns a Retriever for this managed index."""
+        return self._index.as_retriever(**kwargs)
+
+    def as_query_engine(
+        self,
+        *,
+        temperature: float = 0.7,
+        answer_style: Any = 1,
+        safety_setting: List[Any] = [],
+        **kwargs: Any,
+    ) -> BaseQueryEngine:
+        """Returns the AQA engine for this index.
+
+        Example:
+          query_engine = index.as_query_engine(
+              temperature=0.7,
+              answer_style=AnswerStyle.ABSTRACTIVE,
+              safety_setting=[
+                  SafetySetting(
+                      category=HARM_CATEGORY_SEXUALLY_EXPLICIT,
+                      threshold=HarmBlockThreshold.BLOCK_LOW_AND_ABOVE,
+                  ),
+              ]
+          )
+
+        Args:
+            temperature: 0.0 to 1.0.
+            answer_style: See `google.ai.generativelanguage.GenerateAnswerRequest.AnswerStyle`
+            safety_setting: See `google.ai.generativelanguage.SafetySetting`.
+
+        Returns:
+            A query engine that uses Google's AQA model. The query engine will
+            return a `Response` object.
+
+            `Response`'s `source_nodes` will begin with a list of attributed
+            passages. These passages are the ones that were used to construct
+            the grounded response. These passages will always have no score,
+            the only way to mark them as attributed passages. Then, the list
+            will follow with the originally provided passages, which will have
+            a score from the retrieval.
+
+            `Response`'s `metadata` may also have have an entry with key
+            `answerable_probability`, which is the probability that the grounded
+            answer is likely correct.
+        """
+        # NOTE: lazy import
+        from llama_index.query_engine.retriever_query_engine import RetrieverQueryEngine
```

Comment:
wondering why need lazy import here. bc of import dependency cycle?

Reply:
I'm not sure too. I copied it from https://github.com/run-llama/llama_index/blob/771219236021c9cf906c598bb6f750539c791e65/llama_index/indices/base.py#L342-L351